src/creativebuilder/domain/entity/renderer.py

```diff
@@ -0,0 +1,91 @@
+from creativebuilder.domain.entity.template_loader import TemplateLoader
+
+
+class CreativeRenderer:
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def render(self, creative_config: Dict[str, Any], dynamic_content: Dict[str, Any]) -> None:
+        raise NotImplementedError
+
+
+class HTMLCreativeRenderer(CreativeRenderer):
+    def __init__(self, template_loader: TemplateLoader) -> None:
+        self.template_loader = template_loader
+
+    def customize_content(self, content: Dict[str, Any]) -> None:
+        return
+
+    def render(self, creative_config: Dict[str, Any], content: Dict[str, Any]) -> None:
+        self.customize_content(content)
+        template = self.template_loader.load()
+        html = template.render(
+            image_url=content['image_url'],
+            background_image_url=creative_config['background_image_url'],
+            icon=creative_config.get('icon') or DEFAULT_ICON,
+            title=content['name'],
+            product_feed_id=str(content['product_feed_id']),
+            description=content['description'],
+            customized_css=self.template_loader.load_customized_css(),
+        )
+        creative_config['html'] = html
+
+
+class TmonCreativeRenderer(HTMLCreativeRenderer):
+    def __init__(self, template_loader: TemplateLoader) -> None:
+        super().__init__(template_loader)
+
+    def customize_content(self, content: Dict[str, Any]) -> None:
+        content['description'] = '￦{:,}'.format(content['price']) if 'price' in content else '타임커머스 티몬'
+
+
+class HnsCreativeRenderer(HTMLCreativeRenderer):
+    def __init__(self, template_loader: TemplateLoader) -> None:
+        super().__init__(template_loader)
+
+    def customize_content(self, content: Dict[str, Any]) -> None:
+        discount_amount = int(float(content.get('discount_amount') or 0))
+        save_amount = int(float(content.get('save_amount') or 0))
+
+        if discount_amount == HNS_MOBILE_DISCOUNT_INDICATOR or save_amount == HNS_MOBILE_DISCOUNT_INDICATOR:
+            content['description'] = '< 홈&쇼핑 모바일 할인가'
+        else:
+            content['description'] = '< 홈&쇼핑 가격보기'
+
+
+class WMPCreativeRenderer(HTMLCreativeRenderer):
+    def __init__(self, template_loader: TemplateLoader) -> None:
+        super().__init__(template_loader)
+
+    def customize_content(self, content: Dict[str, Any]) -> None:
+        content['descrption'] = ''
+
+
+class NativeCreativeRenderer(CreativeRenderer):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def render(self, creative_config: Dict[str, Any], dynamic_content: Dict[str, Any]) -> None:
+        resize_adapter = ImageResizeAdapter()
+        creative_config['image_url'] = resize_adapter.to_native(dynamic_content['image_url'])
+        creative_config['description'] = dynamic_content['name']
+
+
+class CreativeRendererFactory:
+    @staticmethod
+    def create(creative_type: int, product: Product) -> Optional[CreativeRenderer]:
```

Comment:
```suggestion
    def create(creative_type: int, product: Product) -> CreativeRenderer:
```
factory는 instance 생성이 목표이므로 optional로 리턴하는 것은 부적절해 보이네요 🤔